future-apps/payroll/app/src/screens/components/PreviousSalary.js

```diff
@@ -107,8 +107,25 @@ const StyledFilters = styled.div`
   }
 `
 
+function getExchangedAmount (payment) {
+  return (payment.amount.amount / payment.exchangeRate.amount)
+}
+
+function parsePayments (payments) {
+  return payments.map((payment) => {
+    const exchanged = getExchangedAmount(payment)
+    return { ...payment, exchanged }
+  })
+}
 
-function mapStateToProps ({ salaryAllocation = [], employees = [], accountAddress = {}, payments = [], denominationToken = [] }) {
+function mapStateToProps ({
+  salaryAllocation = [],
+  employees = [],
+  accountAddress = {},
+  payments = [],
+  denominationToken = []
+}) {
+  payments = parsePayments(payments)
```

Comment:
Nice!

Reply:
Anyways in #81 I have refactor this and I put it into store/marshalling.js